src/fzx/tui/color.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include "fzx/macros.hpp"
+#include <cstdint>
+#include <sstream>
+#include <variant>
+#include <fmt/core.h>
+
+namespace fzx {
+constexpr static uint8_t parseNibble(char c)
+{
+  ASSERT((c >= '0' && c <= '9') || (c >= 'a' && c <= 'f'));
+  if (c >= '0' && c <= '9') {
+    return c - '0';
+  } else {
+    return c - 'a' + 0xa;
+  }
+}
+
+constexpr static uint8_t parseByte(std::string_view byte)
+{
+  return parseNibble(byte[0]) * 16 + parseNibble(byte[1]);
+}
+
+enum TermColorCode : uint8_t {
+  Black,
+  Red,
+  Green,
+  Yellow,
+  Blue,
+  Purple,
+  Cyan,
+  White,
```

Comment:
If the values correspond one-to-one with terminal color codes (it's fine if they don't by the way, adding `BrightX` variants would be ok too), then I'd be explicit about values, like `Black = 0`, `Red = 1` etc

Reply:
they do correspond one-on-one, in fact i depend on this in setFg/setBg. I don't want to add bright ones because the codes for bright and normal are as different as fg and bg, so doing just this makes it easier to generalize.

I'll make the values explicit

Reply:
Still though, the default color (`\e[39m`) should be represented somehow, because it's not necessarily the same as white/`\e[37m`

Reply:
won't clearColor work if you want to reset it to default?

Reply:
Yes, but you still might want to use the default color in a theme. For me the default foreground text color is unique, and white/bright white is brighter than the default color.

Reply:
ah - is `\e[49m` for setting the background to default then? if yes then it's good, i can't find documentation for "default"

Reply:
Yup, 39 is the default fg, 49 default bg

Reply:
done